src/receive_imf.rs

```diff
@@ -2391,12 +2376,31 @@ async fn apply_group_changes(
                     &new_members,
                 )
                 .await?;
-                chat_contacts = new_members;
                 send_event_chat_modified = true;
             }
         }
     }
 
+    let new_chat_contacts = HashSet::<ContactId>::from_iter(
+        chat::get_chat_contacts(context, chat_id)
+            .await?
+            .iter()
+            .copied(),
+    );
+
+    // These are for adding info messages about implicit membership changes, so they are only
+    // filled when such messages are needed.
+    let mut added_ids: HashSet<ContactId> = new_chat_contacts
+        .difference(&chat_contacts)
+        .copied()
+        .collect();
+    let mut removed_ids: HashSet<ContactId> = chat_contacts
+        .difference(&new_chat_contacts)
+        .copied()
+        .collect();
+
+    chat_contacts = new_chat_contacts;
```

Comment:
I think that the logic would be easier to follow by removing this line and using `new_chats_contacts` instead of `chats_contacts` below.